### ISSUE TITLE
Scapy not getting response in multithreaded app

### ISSUE THREAD

**Issue by rafajot** [secdev/scapy#778]
I'm wondering if there are any known issues with the scapy being used in multi-threaded applications. 
Please refer to the attached scapy_multithreaded.py application. It has two functions: rtt_ip_ICMP(ip) and rtt_ip_UDP(ip) which send ICMP and UDP packets respectively to 127.0.0.1 (also with 127.0.0.1 srcip).
They work fine in single-thread application but in case of multi-threaded app lot of responses are missing.
Attaching python script as well as the output from command:
`./scapy_multithreaded.py > scapy_multithreaded.txt`

Scapy: 2.3.3 
Linux 2.6.32-573.3.1.el6.x86_64 #1 SMP Thu Aug 13 22:55:16 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
Python 2.7.6

BTW I noticed that the workaround for this issue might be changing in load_ips() function, line 41 ThreadPoolExecutor to ProcessPoolExecutor. Anyway I have a strong impression this is an indication of a bug in scapy.

[scapy_multithreaded.zip](https://github.com/secdev/scapy/files/1246826/scapy_multithreaded.zip)

**Comment by p-l-** [comment]
I cannot reproduce your bug. Can you try with the current development version?

**Comment by rafajot** [comment]
Yes, I'm having this issue with current dev version too:

![image](https://user-images.githubusercontent.com/2522652/29669630-625c0d6a-88e4-11e7-9da6-db47e4ae01a7.png)

**Comment by p-l-** [comment]
OK, can you summarize clearcly with which OS, architecture, Python version and Scapy version combination you do and do not experience this issue?

Please avoid screenshots and prefer code blocks to format your messages, that's much more readable.

**Comment by rafajot** [comment]
All right, I have retested all platforms available for me and in every case I used the current dev scapy version: 2.3.3.dev577

# Platform1: No issues at all
Linux 3.13.0-65-generic #106-Ubuntu SMP Fri Oct 2 22:08:27 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
Python 2.7.6 as well as Python 2.7.13

# Platform 2: occasional failures (usually ~4-6 out of 20) on UDP layer only
Linux 3.2.0-4-amd64 #1 SMP Debian 3.2.60-1+deb7u3 x86_64 GNU/Linux
Python 2.7.3

# Platform 3: Majority of failures (~75%) for both UDP and ICMP
Linux 2.6.32-573.3.1.el6.x86_64 #1 SMP Thu Aug 13 22:55:16 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
Python 2.7.13

**Comment by p-l-** [comment]
I've added my test platform, which would also suggest a kernel-related issue, fixed somewhere between 3.2 and 3.13. I will try to find other platforms to run this test.

**Comment by guedou** [comment]
Shall we close this ?

**Comment by rafajot** [comment]
I guess it depends if that's a scapy issue or kernel issue. Unfortunately I don't know answer to that question. In addition if there is no planned activity around this case I would say it's probably also good enough reason to close it.

**Comment by p-l-** [comment]
Probably kernel related, please reopen if needed. Please reopen if you face this issue with a kernel **more recent than 3.13**.
